services/common/rs_server_common/data_retrieval/eodag_provider.py

```diff
@@ -0,0 +1,80 @@
+"""EODAG Provider."""
+from pathlib import Path
+
+from eodag import EODataAccessGateway, EOProduct
+
+from .provider import CreateProviderFailed, Product, Provider, TimeRange
+
+
+class EodagProvider(Provider):
+    """An EODAG provider.
+
+    It uses EODAG to provide data from external sources.
+    """
+
+    def __init__(self, config_file: str, provider: str):
+        """Create a EODAG provider.
+
+        :param config: the eodag configuration
+        """
+        self.provider: str = provider
+        self.client: EODataAccessGateway = self.init_eodag_client(config_file)
+        self.client.set_preferred_provider(self.provider)
+
+    def init_eodag_client(self, config_file: Path):
```

Comment:
there is a unconsistency there since you're using a str rather than a path.
The given unit test is probably wrong now.

Reply:
Fixed.